libs/platform/src/lib/components/table/table.component.ts

```diff
@@ -211,60 +382,374 @@ export class TableComponent implements OnChanges, OnInit, AfterViewInit, OnDestr
+        this._popoverOpen = false;
+    }
+
+    /** @hidden */
+    filter(field: string, value: string): void {
+        this._tableService.filter([
+            {
+                field: field,
+                value: value,
+                strategy: CollectionStringFilterStrategy.CONTAINS
+            } as CollectionFilter
+        ]);
+
+        this._popoverOpen = false;
+    }
+
+    /** @hidden */
+    group(field: string): void {
+        this._tableService.group(field, SortDirection.ASC);
+        this._popoverOpen = false;
+    }
+
+    /** @hidden */
+    freezeTo(columnKey: string): void {
+        this._tableService.freezeTo(columnKey);
+        this.freezeColumnsTo = columnKey;
+        this._setFreezableInfo();
+    }
+
+    /** @hidden */
+    unfreeze(columnKey: string): void {
+        const idx = this._freezableColumns.indexOf(columnKey);
+        const freezeToKey = this._freezableColumns[idx - 1];
+        this._tableService.freezeTo(freezeToKey);
+        this.freezeColumnsTo = freezeToKey;
+        this._setFreezableInfo();
+    }
+
+    /** Search */
+    search(searchInput: SearchInput): void {
+        this._tableService.search(searchInput);
+    }
+
+    /** Open Sort Dialog */
+    openSortingDialog(): void {
+        const state = this.getTableState();
+        const columns = this._getTableColumnsAsArray();
+        const sortBy = state.sortBy?.[0];
+        const dialogData: SortDialogData = {
+            columns: columns.filter(({ sortable }) => sortable),
+            direction: sortBy?.direction,
+            field: sortBy?.field
+        };
+
+        const dialogRef = this._dialogService.open(SortingComponent, {
+            ...dialogConfig,
+            data: dialogData
+        });
+
+        this._subscriptions.add(
+            dialogRef.afterClosed.pipe(filter((result) => !!result)).subscribe((result: SortDialogResultData) => {
+                this._tableService.sort(result.field, result.direction);
+            })
+        );
+    }
+
+    /** Open Filtering Dialog */
+    openFilteringDialog(): void {
+        const state = this.getTableState();
+        const columns = this._getTableColumnsAsArray();
+        const dialogData: FiltersDialogData = {
+            columns: columns,
+            viewSettingsFilters: this.viewSettingsFilters,
+            filterBy: state?.filterBy
+        };
+
+        const dialogRef = this._dialogService.open(FiltersComponent, {
+            responsivePadding: false,
+            verticalPadding: false,
+            minWidth: '30%',
+            minHeight: '50%',
+            data: dialogData
+        } as DialogConfig);
+
+        this._subscriptions.add(
+            dialogRef.afterClosed.pipe(filter((result) => !!result)).subscribe((result: FiltersDialogResultData) => {
+                if (!result.filterBy) {
+                    return;
+                }
+                this._tableService.filter(result.filterBy);
+            })
```

Comment:
Maybe let's simplify it to:
```suggestion
            dialogRef.afterClosed.pipe(filter(result => !!result?.filterBy).subscribe((result: FiltersDialogResultData) => 
                this._tableService.filter(result.filterBy)
            )
```